src/pkg/packager/network.go

```diff
@@ -178,7 +171,32 @@ func (p *Packager) handleOciPackage() error {
 		return nil
 	}
 	copyOpts.PostCopy = copyOpts.OnCopySkipped
+	isPartialPull := len(components) > 0
+	if isPartialPull {
+		alwaysPull := []string{"zarf.yaml", "checksums.txt", "zarf.yaml.sig"}
+		components = append(components, alwaysPull...)
+		copyOpts.FindSuccessors = func(ctx context.Context, fetcher content.Fetcher, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
+			nodes, err := content.Successors(ctx, fetcher, desc)
+			if err != nil {
+				return nil, err
+			}
+			var ret []ocispec.Descriptor
+			for _, node := range nodes {
+				if utils.SliceContains(components, node.Annotations[ocispec.AnnotationTitle]) {
+					ret = append(ret, node)
+				} else {
+					estimatedBytes -= node.Size
```

Comment:
We should have this logic be computed in the original `getOCIPackageSize()` function